qualtran/bloqs/qsp/generalized_qsp.py

```diff
@@ -382,14 +374,15 @@ def decompose_from_registers(
         num_inverse_applications = self.negative_power
 
         yield self.signal_rotations[0].on(signal_qubit)
+        flat_qubits_for_u = merge_qubits(self.U.signature, **quregs)
         for signal_rotation in self.signal_rotations[1:]:
             if num_inverse_applications > 0:
                 # apply C-U^\dagger
-                yield self.U.adjoint().on_registers(**quregs).controlled_by(signal_qubit)
+                yield self.U.adjoint().controlled().on(signal_qubit, *flat_qubits_for_u)
                 num_inverse_applications -= 1
             else:
                 # apply C[0]-U
-                yield self.U.on_registers(**quregs).controlled_by(signal_qubit, control_values=[0])
+                yield self.U.controlled(control_values=[0]).on(signal_qubit, *flat_qubits_for_u)
```

Comment:
Should we default to using qualtran style controls `CtrlSpec(cvs=0)` instead of cirq style `control_values=[0]`?

Reply:
Reverted this change because `self.U.on_registers(**quregs).controlled_by(signal_qubit)` now uses the `Controlled` bloq instead of cirq gate. The `op.controlled_by` still expects the cirq-style API since we don't have a custom `GWROperation` where we can override the `controlled_by` to expect the bloq-style API as well. But I think it's fine right now to do things this way